ompi/mpi/c/type_create_hindexed.c.in

```diff
@@ -0,0 +1,85 @@
+ * Copyright (c) 2004-2020 The University of Tennessee and The University
+ *                         of Tennessee Research Foundation.  All rights
+ *                         reserved.
+ * Copyright (c) 2004-2008 High Performance Computing Center Stuttgart,
+ *                         University of Stuttgart.  All rights reserved.
+ * Copyright (c) 2004-2005 The Regents of the University of California.
+ *                         All rights reserved.
+ * Copyright (c) 2013      Los Alamos National Security, LLC.  All rights
+ *                         reserved.
+ * Copyright (c) 2015      Research Organization for Information Science
+ *                         and Technology (RIST). All rights reserved.
+ * Copyright (c) 2024      Triad National Security, LLC. All rights
+ *                         reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "ompi_config.h"
+
+#include "ompi/mpi/c/bindings.h"
+#include "ompi/runtime/params.h"
+#include "ompi/communicator/communicator.h"
+#include "ompi/errhandler/errhandler.h"
+#include "ompi/datatype/ompi_datatype.h"
+#include "ompi/memchecker.h"
+
+
+PROTOTYPE ERROR_CLASS type_create_hindexed(COUNT count,
+                                           COUNT_ARRAY array_of_blocklengths,
+                                           AINT_COUNT_ARRAY array_of_displacements,
+                                           DATATYPE oldtype,
+                                           DATATYPE_OUT newtype)
+{
+    int rc, i;
+
+    MEMCHECKER(
+        memchecker_datatype(oldtype);
+        );
+
+    if( MPI_PARAM_CHECK ) {
+        OMPI_ERR_INIT_FINALIZE(FUNC_NAME);
+        if( count < 0 ) {
+            return OMPI_ERRHANDLER_NOHANDLE_INVOKE(MPI_ERR_COUNT,
+                                          FUNC_NAME);
+        } else if ((count > 0) && (NULL == array_of_blocklengths ||
+                                   NULL == array_of_displacements)) {
+            return OMPI_ERRHANDLER_NOHANDLE_INVOKE(MPI_ERR_ARG,
+                                          FUNC_NAME);
+        } else if (MPI_DATATYPE_NULL == oldtype || NULL == oldtype ||
+                   NULL == newtype) {
+            return OMPI_ERRHANDLER_NOHANDLE_INVOKE(MPI_ERR_TYPE,
+                                          FUNC_NAME);
+        }
+        for ( i = 0; i < count; i++ ) {
+            if (array_of_blocklengths[i] < 0) {
+                return OMPI_ERRHANDLER_NOHANDLE_INVOKE(MPI_ERR_ARG,
+                                              FUNC_NAME );
+            }
+        }
+    }
+
+    rc = ompi_datatype_create_hindexed( count, array_of_blocklengths, array_of_displacements,
+                                        oldtype, newtype );
+    if( rc != MPI_SUCCESS ) {
+        ompi_datatype_destroy( newtype );
+        OMPI_ERRHANDLER_NOHANDLE_RETURN( rc, rc, FUNC_NAME );
+    }
+    /* data description */
+    {
+        const int* a_i[2] = {&count, array_of_blocklengths};
```

Comment:
This line creates a compiler warning about mismatched types.

Reply:
resolved with temporary patch to be removed once we embiggen the datatype engine